chore(tasks): Update `prettier_conformance` task

leaysgur opened the pull request:
- Update Prettier version: `3.3.3` > `3.4.1`
- Add more ignored tests
- Add `jsx` tests
- Remove unused `runner.rs`
- Refactoring

Boshen commented:
### Merge activity

* **Nov 27, 3:40 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
chore(tasks): Update `prettier_conformance` task (#7500)

- Update Prettier version: `3.3.3` > `3.4.1`
- Add more ignored tests
- Add `jsx` tests
- Remove unused `runner.rs`
- Refactoring
